queries/haskell/highlights.scm

```diff
@@ -1,14 +1,65 @@
 ;; ----------------------------------------------------------------------------
-;; Parameters
+;; Parameters and variables
 
 ;; NOTE: These are at the top, so that they have low priority,
 ;; and don't override destructured parameters 
 
+(variable) @variable
+(pat_wildcard) @variable
+
 (function
-  patterns: (patterns (pat_name) @parameter))
+  patterns: (patterns [
+    (pat_name (variable) @parameter)
+    (_ (pat_name (variable) @parameter))
+    (_ (pat_apply (pat_name (variable) @parameter)))
+    (_ (pat_typed pattern: (pat_name) @parameter))
+    (_ (pat_typed pattern: (pat_apply (pat_name (variable) @parameter))))
```

Comment:
this query seems too nested to me. What stops it from being a separate `(pat_name (variable) @parameter)`. All these patterns seem to be relying on that.

@amaanq what's your thoughts about it

Reply:
If I just query `(pat_name (variable) @parameter)`, it'll match any pattern matching expressions, for example

```haskell
case mFoo of
  Just bar -> -- <...>
  Nothing -> -- <...>
```
which would require equally nested queries.

Reply:
🤔 come to think of it, maybe it's better not to try to query for destructured patterns, because those can be arbitrarily nested, e.g.

```haskell
func (Left (Just a)) = -- <...>
func (Right (Just a)) = -- <...>
```

Reply:
Surprisingly, matching the entire expression appears to work for highlighting.

![2023-10-05_07-00](https://github.com/nvim-treesitter/nvim-treesitter/assets/12857160/5e876c22-1dcb-41fe-808c-d1dbe9284412)

Reply:
> maybe it's better not to try to query for destructured patterns, because those can be arbitrarily nested, e.g.

I'd say just avoid it entirely. If you nested twice and it works, and then the 3rd nest doesn't work and people will randomly ask why doesn't this work

And then there will be some edge cases like https://github.com/projekt0n/github-nvim-theme/pull/250/files, which I do not want to happen at all

Reply:
I've just updated it to match the entire expression, and it works (see the screen shot).
I have my theme configured to italicize `@parameter` matches.